common.go

```diff
@@ -756,17 +751,30 @@ func (c *Config) time() time.Time {
 	return t()
 }
 
-func (c *Config) cipherSuites(version uint16) []uint16 {
-	if version >= VersionTLS13 {
-		s := c.TLS13CipherSuites
-		if s == nil {
-			s = defaultTLS13CipherSuites()
+func hasOverlappingCipherSuites(cs1, cs2 []uint16) bool {
+	for _, c1 := range cs1 {
+		for _, c2 := range cs2 {
+			if c1 == c2 {
+				return true
+			}
 		}
-		return s
 	}
+	return false
+}
+
+func (c *Config) cipherSuites() []uint16 {
 	s := c.CipherSuites
 	if s == nil {
 		s = defaultCipherSuites()
+	} else if c.maxVersion() >= VersionTLS13 {
+		// Ensure that TLS 1.3 suites are always present, but respect
+		// the application cipher suite preferences.
+		s13 := defaultTLS13CipherSuites()
+		if !hasOverlappingCipherSuites(s, s13) {
+			allSuites := make([]uint16, len(s13)+len(s))
+			allSuites = append(allSuites, s13...)
```

Comment:
I wonder if append(allSuites, s13..., s...) will save us one line of code and make it more apparent that we push s13 ahead of s)

Reply:
I tried that at first, but it is not supported: https://golang.org/ref/spec#Passing_arguments_to_..._parameters

Reply:
👍 yeah, I was going to try that myself.